### PR TITLE
#120: API/core/atomics/atomic_fetch_op from md to rst

### PR DESCRIPTION
**Pull request by pierrepebay**
* API/core/atomics/atomic_fetch_[op] transition from .md to .rst

before (.md) | after (.rst)
:----------------: | :-------------:
image | image
image | ---
image | image
image  | image

### CODE REVIEW

**Review by fnrizzi — CHANGES_REQUESTED** [review]
as per our 1:1 discussion

**Review by fnrizzi — APPROVED** [review]

File: docs/source/API/core/atomics/atomic_fetch_op.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+Usage
+-----
+
+.. code-block:: cpp
+
+   old_value =  atomic_fetch_[op](ptr_to_value, update_value);
+
+Atomically updates the variable at the address given by ``ptr_to_value`` with ``update_value``
+according to the relevant operation, and returns the previous value found at that address.
```
Comment (rgayatri23) [inline docs/source/API/core/atomics/atomic_fetch_op.rst:17]: Should we say "according to the relevant operation `op` "

File: docs/source/API/core/atomics/atomic_fetch_op.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+Header File: <Kokkos_Core.hpp>
+
+Usage
+-----
+
+.. code-block:: cpp
+
+   old_value =  atomic_fetch_[op](ptr_to_value, update_value);
+
+Atomically updates the variable at the address given by ``ptr_to_value`` with ``update_value``
```
Comment (ajpowelsnl) [inline docs/source/API/core/atomics/atomic_fetch_op.rst:16]: Is it correct that this `atomic_fetch_add` (for instance) returns the previous value at that address vs. the previous values.

File: docs/source/API/core/atomics/atomic_fetch_op.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+-----------
+
+.. cppkokkos:function:: template<class T> T atomic_fetch_add(T* const ptr_to_value, const T value);
+
+   Atomically executes ``tmp = *ptr_to_value; *ptr_to_value += value; return tmp;``.
+
+   :param ptr_to_value: address of the value to be updated
+   :param value: value to be added
+
+.. cppkokkos:function:: template<class T> T atomic_fetch_and(T* const ptr_to_value, const T value);
```
Comment (ajpowelsnl) [inline docs/source/API/core/atomics/atomic_fetch_op.rst:29]: Typo?  Should `atomic_fetch_and` (sic) be `atomic_fetch_add`?

**Review by ajpowelsnl — APPROVED** [review]
A possible typo mentioned, but otherwise fine.

File: docs/source/API/core/atomics/atomic_fetch_op.rst (hunk 1 above)
Comment (fnrizzi) [reply docs/source/API/core/atomics/atomic_fetch_op.rst:17]: fixed

File: docs/source/API/core/atomics/atomic_fetch_op.rst (hunk 3 above)
Comment (fnrizzi) [reply docs/source/API/core/atomics/atomic_fetch_op.rst:29]: no, this is doing ``&=`` not ``+=``

**Pull request merged**